### ISSUE TITLE
Load Trainer state

### ISSUE THREAD

**Issue by Aktsvigun** [huggingface/transformers#12529]
# 🚀 Feature request

Hi,
I wonder if there is a method to load a saved state of the trainer, so that I can continue the training loop from where I started.
If I understand it right, I need to load the saved states of _model_, _optimizer_ and _scheduler_. 
## Motivation

However, it would be much more convenient to have all of this done in one method, when you only pass the `results` folder path to the method.

## Your contribution

In case there is not, I can submit the PR, since it does not seem too hard to be prepared.

**Comment by LysandreJik** [comment]
cc @sgugger

**Comment by sgugger** [comment]
How would this be different from
```
trainer.train(resume_from_checkpoint=True)
```
?

**Comment by Aktsvigun** [comment]
That's pretty similar, still, when one needs to use `trainer.predict` of a trained model, for instance.

**Comment by sgugger** [comment]
You don't need the optimizer and scheduler states for this, just the model.

**Comment by Aktsvigun** [comment]
This makes sense, thank you! Closing the issue.
